Drivers/IMU_DRIVER/IMU.h

```diff
@@ -0,0 +1,88 @@
+/*
+    LSM6DSOXTR IMU DRIVER Header File
+    Link to part Datasheet for reference:
+    https://www.st.com/resource/en/datasheet/lsm6dsox.pdf
+
+    Author: Hamza Iqbal
+*/
+
+#ifndef IMU_DRIVER_H
+#define IMU_DRIVER_H
+#include "stm32f4xx_hal.h"
+
+#define IMU_I2C_ADDR        0x35
+// Not sure if these are all needed, also not sure if more need ot be added
+// For register descriptions reference datasheet pages 47 - 98
+#define FUNC_CFG_ACCESS     0x01    // Enagle embedded functions register
+#define INT1_CTRL           0x0D    // INT1 pin control, used for interrupts
+#define INT2_CTRL           0x0E    // INT2 pin control, used for interrupts
```

Comment:
I support the ALL_CAPS for macros but make them more verbose and also prefix them with CHIPNAME_REG_XXXX so we can avoid namespace issues